### PR TITLE
raftstore: fix split buckets bug

### LINKED ISSUE #12467: [Dynamic Regions] tikv panic after drop big table
### What version of TiKV are you using?
./tikv-server -V
TiKV
Release Version: 6.0.0-alpha
Edition: Community
Git Commit Hash: https://github.com/tikv/tikv/commit/95c4d0ed36369c254b1bee36eb007004ee834948
Git Commit Branch: heads/refs/tags/v6.1.0-nightly
UTC Build Time: 2022-04-21 10:58:55
Rust Version: rustc 1.60.0-nightly (1e12aef3f 2022-02-13)
Enable Features: jemalloc mem-profiling portable sse test-engines-rocksdb cloud-aws cloud-gcp cloud-azure
Profile: dist_release
2022-04-24T10:07:55.070+0800 INFO k8s/client.go:107 it should be noted that a long-running command will not be interrupted even the use case has ended. For more information, please refer to https://github.com/pingcap/test-infra/discussions/129
./pd-server -V
Release Version: v6.1.0-alpha-40-g1d8eba0
Edition: Community
Git Commit Hash: 1d8eba08d9907ef1a6fdebd1ebf6b218b9338d91
Git Branch: feature/region_monitor
UTC Build Time: 2022-04-22 12:21:42
2022-04-24T10:07:55.245+0800 INFO k8s/client.go:107 it should be noted that a long-running command will not be interrupted even the use case has ended. For more information, please refer to https://github.com/pingcap/test-infra/discussions/129
./tidb-server -V
Release Version: v6.1.0-nightly
Edition: Community
Git Commit Hash: 571d97bb16f6f69fcbcf74648380da3c2e80a3fb
Git Branch: heads/refs/tags/v6.1.0-nightly
UTC Build Time: 2022-04-20 23:02:41
GoVersion: go1.18
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
2022-04-24T10:07:55.385+0800 INFO k8s/client.go:107 it should be noted that a long-running command will not be interrupted even the use case has ended. For more information, please refer to https://github.com/pingcap/test-infra/discussions/129

### What operating system and CPU are you using?
8core、16G
enable-region-bucket = true
region-bucket-size = "128M"
region-max-keys = 144000000000
region-max-size = "1.3GB"
region-split-keys = 96000000000
region-split-size = "1GB"

### Steps to reproduce
[cmd="sysbench --mysql-user=root --db-driver=mysql --time=0 --report-interval=10 --mysql-db=sysbench_64_7000w --mysql-host=tc-tidb.glh-dynamic-regions-jw8pv --mysql-port=4000 --threads=512 --tables=64 --table-size=70000000 --mysql-ignore-errors=2013,1213,1105,1205,8022,8027,8028,9004,9007 --rand-type=special oltp_insert run"] [nodename=benchtoolset]
[2022/05/02 12:57:22.814 +08:00] [INFO] 
after running 3days+，drop two big database 
image

### What did you expect?
all pods are normal

### What did happened?
tikv-0 panic
level="FATAL" log="[lib.rs:468] [\"assertion failed: `(left == right)`\\n  left: `1`,\\n right: `6`\"] [backtrace=\"   0: tikv_util::set_panic_hook::{{closure}}\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tikv_util/src/lib.rs:467:18\\n   1: std::panicking::rust_panic_with_hook\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/panicking.rs:702:17\\n   2: std::panicking::begin_panic_handler::{{closure}}\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/panicking.rs:588:13\\n   3: std::sys_common::backtrace::__rust_end_short_backtrace\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/sys_common/backtrace.rs:138:18\\n   4: rust_begin_unwind\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/panicking.rs:584:5\\n   5: core::panicking::panic_fmt\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/core/src/panicking.rs:143:14\\n   6: core::panicking::assert_failed_inner\\n   7: core::panicking::assert_failed\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/core/src/panicking.rs:182:5\\n   8: raftstore::store::worker::split_check::Runner<E,S>::scan_split_keys::{{closure}}\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore/src/store/worker/split_check.rs:432:21\\n      core::result::Result<T,E>::map\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/core/src/result.rs:756:25\\n      raftstore::store::worker::split_check::Runner<E,S>::scan_split_keys\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore/src/store/worker/split_check.rs:368:9\\n   9: raftstore::store::worker::split_check::Runner<E,S>::check_split_and_bucket\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore/src/store/worker/split_check.rs:279:23\\n      <raftstore::store::worker::split_check::Runner<E,S> as tikv_util::worker::pool::Runnable>::run\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore/src/store/worker/split_check.rs:494:18\\n      tikv_util::worker::pool::Worker::start_impl::{{closure}}\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tikv_util/src/worker/pool.rs:450:25\\n      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/core/src/future/mod.rs:84:19\\n      yatp::task::future::RawTask<F>::poll\\n             at rust/git/checkouts/yatp-e704b73c3ee279b6/5f3d580/src/task/future.rs:59:9\\n  10: yatp::task::future::TaskCell::poll\\n             at rust/git/checkouts/yatp-e704b73c3ee279b6/5f3d580/src/task/future.rs:103:9\\n      <yatp::task::future::Runner as yatp::pool::runner::Runner>::handle\\n             at rust/git/checkouts/yatp-e704b73c3ee279b6/5f3d580/src/task/future.rs:387:20\\n  11: <tikv_util::yatp_pool::YatpPoolRunner<T> as yatp::pool::runner::Runner>::handle\\n             at home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/tikv_util/src/yatp_pool/mod.rs:104:24\\n      yatp::pool::worker::WorkerThread<T,R>::run\\n             at rust/git/checkouts/yatp-e704b73c3ee279b6/5f3d580/src/pool/worker.rs:48:13\\n      yatp::pool::builder::LazyBuilder<T>::build::{{closure}}\\n             at rust/git/checkouts/yatp-e704b73c3ee279b6/5f3d580/src/pool/builder.rs:114:25\\n      std::sys_common::backtrace::__rust_begin_short_backtrace\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/sys_common/backtrace.rs:122:18\\n  12: std::thread::Builder::spawn_unchecked_::{{closure}}::{{closure}}\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/thread/mod.rs:498:17\\n      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/core/src/panic/unwind_safe.rs:271:9\\n      std::panicking::try::do_call\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/panicking.rs:492:40\\n      std::panicking::try\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/panicking.rs:456:19\\n      std::panic::catch_unwind\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/panic.rs:137:14\\n      std::thread::Builder::spawn_unchecked_::{{closure}}\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/thread/mod.rs:497:30\\n      core::ops::function::FnOnce::call_once{{vtable.shim}}\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/core/src/ops/function.rs:227:5\\n  13: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/alloc/src/boxed.rs:1854:9\\n      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/alloc/src/boxed.rs:1854:9\\n      std::sys::unix::thread::Thread::new::thread_start\\n             at rustc/1e12aef3fab243407f9d71ba9956cb2a1bf105d5/library/std/src/sys/unix/thread.rs:108:17\\n  14: <unknown>\\n  15: clone\\n\"] [location=/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tikv/components/raftstore/src/store/worker/split_check.rs:432] [thread_name=background-

### PR DESCRIPTION
**Pull request by tonyxuqqi**
Signed-off-by: qi.xu <tonxuqi@outlook.com>

### What is changed and how it works?
Issue Number: Close #12467

What's Changed:

```
It's possible that when the split bucket task runs, the data of end ranges in bucket_range_list have been deleted.
As a result, the buckets generated by scan may miss some bucket ranges. The fix is to add the empty Bucket instances.

Also change the log level to debug for some chatty logs.
```

### Check List 

Tests 

- Unit test

Side effects

### Release note 
```release-note
None
```

### CODE REVIEW

File: components/raftstore/src/store/worker/split_check.rs [hunk 1]
```diff
@@ -430,6 +430,11 @@ where
                 || bucket_range_list.is_empty() && !skip_check_bucket
             {
                 buckets.push(bucket);
+                 // in case some range's data in bucket_range_list is deleted
+                if buckets.len() < bucket_range_list.len() {
```
Comment (zhangjinpeng1987) [inline components/raftstore/src/store/worker/split_check.rs:434]: please add UT to cover it.

Comment (tonyxuqqi) [reply components/raftstore/src/store/worker/split_check.rs:434]: done.

**Review by 5kbpers — APPROVED** [review]

**Review by BusyJay — APPROVED** [review]

**Pull request merged**